crates/vizia_core/src/modifiers/text.rs

```diff
@@ -1,25 +1,37 @@
 use super::internal;
-use crate::{prelude::*, text::Selection};
+use crate::prelude::*;
+use cosmic_text::{Attrs, FamilyOwned, Weight};
 
 /// Modifiers for changing the text properties of a view.
 pub trait TextModifiers: internal::Modifiable {
     /// Sets the text content of the view.
     fn text<U: ToString>(mut self, value: impl Res<U>) -> Self {
         let entity = self.entity();
         value.set_or_bind(self.context(), entity, |cx, entity, val| {
-            if let Some(prev_data) = cx.style.text.get(entity) {
-                if prev_data != &val.to_string() {
-                    cx.style.text.insert(entity, val.to_string());
-
-                    cx.need_relayout();
-                    cx.need_redraw();
-                }
-            } else {
-                cx.style.text.insert(entity, val.to_string());
-
-                cx.need_relayout();
-                cx.need_redraw();
-            }
+            let text_data = val.to_string();
+
+            cx.cosmic_context.with_buffer(entity, |buffer| {
+                buffer.set_text(&text_data, Attrs::new());
+            });
+
+            cx.need_relayout();
+            cx.need_redraw();
+
+            // if let Some(prev_data) = cx.style.text.get(entity) {
+            //     if prev_data != &text_data {
+            //         cx.style.text.insert(entity, text_data);
+
+            //         cx.cosmic_context.with_buffer(entity, f)
+
+            //         cx.need_relayout();
+            //         cx.need_redraw();
+            //     }
+            // } else {
+            //     cx.style.text.insert(entity, text_data);
+
+            //     cx.need_relayout();
+            //     cx.need_redraw();
+            // }
```

Comment:
I know it was me that commented this out but you can go ahead and delete it now.